### DISCUSSION TITLE
How to retrieve the metadata of a file stored in Amazon s3

### DISCUSSION THREAD

**Discussion by Juanjo-GEx** [directus/directus#15591] [Archive]
A file of type *file object* has been stored in Amazon's s3 with the metadata corresponding to its redirection:

```console
{
    "data": {
          "id": "f7b5079a-1615-4cff-a4a0-c54fc9510aa6",
          "storage": "s3",
          "filename_disk": "f7b5079a-1615-4cff-a4a0-c54fc9510aa6.txt",
          "filename_download": "fyemv.txt",
          "title": "fyemv",
          "type": "text/plain",
          "folder": null,
          "uploaded_by": "cc74473f-eff8-4219-9b0f-c9c1e2cf6ac6",
          "uploaded_on": "2022-09-14T11:41:51",
          "modified_by": null,
          "modified_on": "2022-09-14T13:41:51",
          "charset": null,
          "filesize": 0,
          "width": null,
          "height": null,
          "duration": null,
          "embed": null,
          "description": null,
          "location": null,
          "tags": null,
          "metadata": {
                "x-amz-website-redirect-location": "https://www.globalexchange.es/fyemv"
          }
    }
}
```

When trying to retrieve that metadata from the file hosted in an Amazon bucket, they do not appear:

![image](https://user-images.githubusercontent.com/110392472/190381939-0c85181d-08c4-4fe0-a0f2-1592fb834b22.png)

Where and how could I retrieve that metadata?
Thanks a lot!!

**Comment by br41nslug** [comment]
That metadata is saved in the database in the `directus_files` table.

**Comment by Juanjo-GEx** [comment]
Yes, I can obtain the metadata of the file in Directus, but what I don't know how to do is how to obtain that metadata of that object in **Amazon's s3, not in Directus**.

What I want is to create a redirection from Amazon.
[Official documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/how-to-page-redirect.html)

**Reply by rijkvanzanten** [reply]
Directus doesn't currently read / return that metadata from S3 directly 👍🏻

**Reply by Juanjo-GEx** [reply]
Oh what a pity! :disappointed:
I imagine that the metadata of the file that I establish in Directus, there is no way for them to appear in s3, right?

**Reply by rijkvanzanten** [reply]
That's correct. Directus stores the file metadata in the database, not in S3 👍🏻

**Comment by JonL1** [comment]
Totally unrelated, but I was just using your webapp as I need to get rid of some foreign currency 👍
Gracias!

**Reply by Juanjo-GEx** [reply]
No problem mate! [Write here an email](https://www.globalexchange.es/en/Home/Humans-Resources) and the staff of the corresponding department will be happy to assist you.
